pilot/pkg/security/authz/matcher/string.go

```diff
@@ -27,6 +29,9 @@ func StringMatcher(v string) *matcherpb.StringMatcher {
 
 // StringMatcherRegex creates a regex string matcher for regex.
 func StringMatcherRegex(regex string) *matcherpb.StringMatcher {
+	if features.EnableAuthzRegexMatching && strings.HasPrefix(regex, "regex:") {
```

Comment:
Do we need to check the flag again in StringMatcherRegex? It seems it's already enough that this is already done in StringMatcherWithPrefix and `func (srcNamespaceGenerator) principal`.

Reply:
Yes, thanks! removed.